### ISSUE TITLE
Implement Notification for Chat system

### ISSUE THREAD

**Issue by Roongjin** [2110366-2566-2/pic-keeper#86]
This is an epic issue that will resolve the problem of notifications in the `chat` system.

Here, we leverage the real-time functionality of the web-socket to enhance the UX.

To clarify, consider these scenarios:
- When a user is `online` and there are some messages directed to this user, we will first use client-side rendering to prevent the delay from the database queries, i.e., client-side will have a data structure [a map? with `roomId` as key] to keep track of the `newly-added` notifications and display via `(1)` at the end of the `chat room` name, for instance. This will provide a quick response to the display and UX. 
- Importantly, `front-end` must also detect if the message is directed to the `room` in which the user is currently watching or not. In this case, `NO` notification is required because the message will automatically be displayed via the `web-socket`. Think of the `LINE` application, when we are actively chatting with someone, the application simply just display the message and `NO` notification has been created.

- How to persist the data then? If the application solely provide the previous mechanism for the notification system, when user disconnects from the session, all the notification will be lost. To address this issue, `back-end` will enable and endpoint to `register` a notification. `Front-end` could call this API asynchronously to prevent delay, as the display is already enabled in the previous mechanism.

- Next up, when user `switches room`, `front-end` must remove the `(1)` notation from the `client-side` rendering mechanism to tell the user that they have already viewed the corresponding notification. `Front-end` also needs to call API, provided by the `back-end`, to tell the database that the corresponding notification has been viewed/noticed.
- Finally, when user `disconnects` from the session and `connects` back in, `front-end` must query the persisted notification from the database, for data consistency, and continue with the display. The delay in this phase is `ACCEPTABLE` as it occurs only once -- only when the user connects back in. `LINE` application also has this delay.

To summarise, the `back-end` will provide 3 APIs:
- Registering Notification
- Clear Some Notification
- Retrieve All Remaining Notification

If you have any discussion about this implementation, please do not hesitate to let me know.
